server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationAllocationIT.java

```diff
@@ -224,6 +241,70 @@ public void testAllocationWithDisruption() throws Exception {
         verifyPerIndexPrimaryBalance();
     }
 
+    /**
+     * Similar to testSingleIndexShardAllocation test but creates multiple indices, multiple nodes adding in and getting
+     * removed. The test asserts post each such event that primary shard distribution is balanced for each index as well as across the nodes
+     * when the PREFER_PRIMARY_SHARD_REBALANCE is set to true
+     */
+    public void testAllocationAndRebalanceWithDisruption() throws Exception {
```

Comment:
Lets ensure we are running this IT for 1K iterations at least to ensure we are not having any flakiness in the behaviour.

Reply:
Ack.